### PR TITLE
[TS] LPS-127109 Segments with both context and model criteria are not correctly evaluated with guest users (2)

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
https://issues.liferay.com/browse/LPS-127109

Originally sent in PR https://github.com/liferay-tango/liferay-portal/pull/834 

Changes:

1. I have split a test, see commit https://github.com/liferay-tango/liferay-portal/commit/f52bdeddc6ae9208448234d977bf5854499a0a79
2. I have created new tests to check the issues with the OR conjuntion, see commit https://github.com/liferay-tango/liferay-portal/commit/90beac67548f824b274d705d83e24ab701c94d0f
3. I have fixed an issue with the OR conjuntion, see commit https://github.com/liferay-tango/liferay-portal/commit/9f1ac878a2c52b608cf7e236de5cae1940caad43

### CODE REVIEW

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

**Comment by jorgediaz-lr** [comment]
ci:test:tango

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java [hunk 1]
```diff
@@ -237,6 +237,38 @@ public void testGetSegmentsEntryIdsWithContextCriterionAndDefaultUser()
...
+			_group.getGroupId(), User.class.getName(), defaultUser.getUserId(),
+			context);
+
+		Assert.assertArrayEquals(new long[0], segmentsEntryIds);
+
+		context = new Context();
+
+		context.put(Context.LANGUAGE_ID, "es");
+
+		segmentsEntryIds = _segmentsEntryProvider.getSegmentsEntryIds(
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java]: As I said before, you should not call the method twice in the same test, please could you create a new test for this specific use case. Let me know if I can help

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java [hunk 2]
```diff
@@ -298,6 +330,51 @@ public void testGetSegmentsEntryIdsWithContextCriterionAndModelCriterion()
...
+			_group.getGroupId(), User.class.getName(), defaultUser.getUserId(),
+			context);
+
+		Assert.assertArrayEquals(new long[0], segmentsEntryIds);
+
+		context = new Context();
+
+		context.put(Context.LANGUAGE_ID, "en");
+
+		segmentsEntryIds = _segmentsEntryProvider.getSegmentsEntryIds(
```
Comment (darquesdev) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java:350]: As I said before, you should not call the method twice in the same test, please could you create a new test for this specific use case. Let me know if I can help

**Comment by darquesdev** [comment]
Hi @jorgediaz-lr, would you mind squashing the commits in just 2: one for the production code and another for the tests?

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java [hunk 3]
```diff
@@ -228,7 +228,22 @@ protected boolean isMember(
 		Criteria.Conjunction contextConjunction = getConjunction(
 			segmentsEntry, Criteria.Type.CONTEXT);
 
+		String modelFilterString = getFilterString(
+			segmentsEntry, Criteria.Type.MODEL);
+
 		if (context != null) {
+			boolean guestUser = false;
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java:235]: We normally name it as _defaultUser_

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java (hunk 2 above)
Comment (jorgediaz-lr) [reply modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java:350]: Ok, but we are going to have all the DefaultUser tests duplicated with and without the `context.put(Context.SIGNED_IN, false);` parameter

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java [hunk 4]
```diff
@@ -240,7 +240,10 @@ protected boolean isMember(
 				guestUser = true;
 			}
 
-			if (guestUser && Validator.isNotNull(modelFilterString)) {
+			if (guestUser &&
```
Comment (darquesdev) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/provider/BaseSegmentsEntryProvider.java:243]: Please, sort the conditions alphabetically if the order does not matter

**Comment by jorgediaz-lr** [comment]
Resent https://github.com/liferay-tango/liferay-portal/pull/838

**Pull request closed without merging**